README.md

```diff
@@ -264,6 +264,7 @@ Mbed TLS is mostly written in portable C99; however, it has a few platform requi
 - `int` and `size_t` must be at least 32 bits wide.
 - The types `uint8_t`, `uint16_t`, `uint32_t` and their signed equivalents must be available.
 - Mixed-endian platforms are not supported.
+- SIZE_MAX must be at least as big as INT_MAX and UINT_MAX.
```

Comment:
I'm not aware of any platform where `size_t` is smaller than `unsigned int`. And I suspect that Mbed TLS would have bugs on such platforms because we don't actively think about promotions in calculations involving `size_t`. Nonetheless, this is technically a requirement change, so we should announce it in the changelog.